### PR TITLE
chore: change migrate logic to make it use directly

### PR DESCRIPTION
**Pull request by ashuralyk**
Closes: #4 

Changes:
1. the previous migrate script file cannot use after pulling repo, so adjust its logic to match the test of Axon endpoint of Forcerelay/Axon
2. panic error message after delegate call of internal functions is hidden, so show them all to optimize contract debug
3. adjust `WriteAcknowledgement` event parameter assignment
4. fix bug in method of query packet sequences in query handler
5. add scripts to send packet to mock transfer module and query packet sequences

### CODE REVIEW

**Comment by ashuralyk** [comment]
@Flouse please take a look at this solidity repo PR

**Review by Flouse — CHANGES_REQUESTED** [review]
Please only use one package manager and commit either the yarn.lock file or the package-lock.json.

File: README.md [hunk 1]
```diff
@@ -3,14 +3,16 @@ ibc-solidity-contract is the EVM-compatible IBC implementation for chain based o
...
-2. Compile: `npx truffle compile --all`
+2. Compile: `npm run compile`
 
 ## Deployment on AXON
 
 Create `.env` in project root and put axon url like
 
 ```
-AXON_RPC_URL=localhost:8000
+AXON_RPC_URL=http://<Ip>:<Port>
```
Comment (Flouse) [inline README.md:13]:
```suggestion
AXON_HTTP_RPC_URL=http://<Ip>:<Port>
```

Comment (Flouse) [reply README.md:13]: ref: https://github.com/synapseweb3/ibc-solidity-contract/pull/16/files#diff-8f7ceaa83eaf3c53b82b8dc646e61e0d66a865cec9da0baa4c19b9833baacd0fR56

File: contracts/core/25-handler/IBCUtil.sol [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        bool success,
+        bytes memory result,
+        string memory prefix
+    ) internal pure {
+        if (!success) {
+            if (result.length == 0) {
+                revert(string.concat(prefix, " delegatecall error"));
+            }
+            assembly {
+                revert(add(32, result), mload(result))
```
Comment (Flouse) [inline contracts/core/25-handler/IBCUtil.sol:15]: don't under stand this line of code.
why add(32, `bytes`) ?

File: contracts/core/25-handler/IBCUtil.sol [hunk 3]
```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+library IBCUtil {
+    function process_delgatecall(
```
Comment (jjyr) [inline contracts/core/25-handler/IBCUtil.sol:5]: This name is confusing, we are actually handling the result of a delegate-call right?
Suggest changing the name to `check_return`.

Comment (ashuralyk) [reply contracts/core/25-handler/IBCUtil.sol:5]: fine, good suggestion

**Review by jjyr — APPROVED** [review]

**Review by Flouse — APPROVED** [review]

**Pull request merged**